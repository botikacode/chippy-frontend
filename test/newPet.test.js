import React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react-native";
import EditPetScreen from "../screens/EditPet";

import { Alert } from "react-native";

test("Renderiza pet Screen correctamente", () => {
  const tree = renderer.create(<EditPetScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Registra bien la mascota", async () => {
  jest.spyOn(Alert, "alert");

  render(<EditPetScreen/>);

  const nombre = screen.getByPlaceholderText("Nombre");
  const descripcion = screen.getByPlaceholderText("Descripción");

  fireEvent.changeText(nombre, "Federico");
  fireEvent.changeText(descripcion, "Hola, soy Federico");

  expect(nombre.props.value).toBe("Federico");
  expect(descripcion.props.value).toBe("Hola, soy Federico");

  // Hace click en Iniciar Sesión
  //fireEvent.press(screen.getByText("Añadir Mascota"));

  expect(Alert.alert)
});