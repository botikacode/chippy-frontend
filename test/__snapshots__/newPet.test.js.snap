// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderiza pet Screen correctamente 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#0094FF",
        "fontSize": 20,
        "marginBottom": 10,
      }
    }
  >
    A침adir Mascota
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "marginVertical": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      error={false}
      errorText=""
      label="Nombre"
      onChangeText={[Function]}
      placeholder="Nombre"
      placeholderTextColor="#576574"
      returnKeyType="next"
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 4,
          "marginTop": 10,
          "padding": 4,
          "width": "70%",
        }
      }
      value=""
    />
    <TextInput
      error={false}
      errorText=""
      label="Descripci칩n"
      onChangeText={[Function]}
      placeholder="Descripci칩n"
      placeholderTextColor="#576574"
      returnKeyType="next"
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 4,
          "marginTop": 10,
          "padding": 4,
          "width": "70%",
        }
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#0094FF",
        "borderRadius": 5,
        "marginBottom": 3,
        "marginTop": 24,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      A침adir Mascota
    </Text>
  </View>
</View>
`;
